frontend/src/components/AddFundsForm.js

```diff
@@ -0,0 +1,164 @@
+            <span className='col col-12 pb1'>Tell us who these funds are from. <span className='AddFundsForm-switch ml1' onClick={::this.switchState}> Keep { defaultUser.name || defaultUser.email } as the donor </span></span>
+            <div className='col col-12 sm-col-12 md-col-6 lg-col-6 pr1'>
+              <label>{i18n.getString('name')}</label>
+              <Input
+                customClass='js-transaction-name'
+                hasError={validationError.name}
+                value={attributes.name}
+                handleChange={name => onChange({name})} />
+            </div>
+            <div className='col col-12 sm-col-12 md-col-6 lg-col-6 pl1'>
+              <label>{i18n.getString('email')}</label>
+              <Input
+                customClass='js-transaction-email'
+                hasError={validationError.email}
+                value={attributes.email}
+                handleChange={email => onChange({email})} />
+            </div>
+          </div>}
+
+        <div className='line1'>funds breakdown (in USD)</div>
+        <div className='clearfix input-container mb2'>
+          <div className='col col-12' >
+            <span className='col col-4'> Funding amount: </span> 
+            <span className='col col-2 AddFundsForm-amount'> {formatCurrency(this.state.amount, collective.currency)}</span>
+          </div>
+          <div className='col col-12'>
+            <span className='col col-4 AddFundsForm-bb'> Host fees ({collective.hostFeePercent}%): </span>
+            <span className='col col-2 AddFundsForm-host-fees'> {formatCurrency(this.state.hostFees, collective.currency)} </span>
+          </div>
+          <div className='col col-12'>
+            <span className='col col-4'> Net funds for collective:</span>
+            <span className='col col-2 AddFundsForm-amount'> {formatCurrency(this.state.amount - this.state.hostFees, collective.currency)} </span>
+          </div>
+        </div>
+
+        <p className='notice'>When the source of funds is someone other than the host, we deduct the host fee. By clicking below, you agree to set aside these funds in your bank account on behalf of the collective. </p>
+     
+        <AsyncButton
+          color='green'
+          inProgress={inProgress}
+          onClick={onSubmit.bind(this)}>
+          {i18n.getString('addFunds')}
+        </AsyncButton>
+
+      </div>
+    )
+  }
+
+  switchState() {
+    const updatedAreFundsFromDefaultUser = !this.props.attributes.fundsFromHost;
+    this.props.onChange({fundsFromHost: updatedAreFundsFromDefaultUser});
+    this.onAmountChange(null, updatedAreFundsFromDefaultUser);
+  }
+
+  onAmountChange(updatedAmountText, areFundsFromDefaultUser) {
+    let amount;
+    if (updatedAmountText) {
+      amount = Math.round(100 * updatedAmountText);
+    } else if (updatedAmountText === '') {
+      amount = 0
+    } else {
+      amount = this.state.amount
+    }
+    const addHostFees = !areFundsFromDefaultUser;
```

Comment:
I was trying to keep the component independent of the idea of host. From it's point of view, there is a default funder to show in UI and the user can decide to change that user.